File: BubbleSort/src/BubbleSort.java

public class BubbleSort {
	
	public static void main(String[] args)
	{
		int n,tmp;
		System.out.println("Bubble Sort!!");
		
		int[] numbers = {5,9,1,2,3,4};// use this to sort in main
		int[] numbers1 = {5,9,1,2,3,4};// give it to method bubbleSort
		
		bubbleSort(numbers1);// return sorted array
		System.out.println(numbers1[2]);
		/* use for loop and sort an array in the main*/
		for(int i =0;i<numbers.length - 2;i++)
		{
			for(int j=0;j<numbers.length - 1;j++)
			{
				if(numbers[j]>numbers[j+1])
				{
					tmp = numbers[j];
					numbers[j] = numbers[j+1];
					numbers[j+1] = tmp;
				}
			}
		
		
		}
		
		System.out.println(numbers[2]);
		
	}
	/* use a bubbleSort method and sort an array*/
	public static int[] bubbleSort(int [] a){
		int len = a.length;
		int tmp1;
		//int[] arr = new int[len];
		
		for(int i =0;i<len - 2;i++)
		{
			for(int j=0;j<len - 1;j++)
			{
				if(a[j]>a[j+1])
				{
					tmp1 = a[j];
					a[j] = a[j+1];
					a[j+1] = tmp1;
				}
			}
		
		
		}
		return a;
	}

}

File: pythonscripts/bubblesort.py
def main():
    list = [5,3,1,0,53,21,331,2,6,8,0]
    print(bubblesort(list))



def bubblesort(numbers):
    print("length=",len(numbers),"list[3]=",numbers[3])
    for i in range(0,len(numbers)):
        for j in range(0,len(numbers)-1):
            if numbers[j] > numbers[j+1]:
                tmp = numbers[j+1]
                numbers[j+1] = numbers[j]
                numbers[j] = tmp
            
   
    return numbers




if __name__=="__main__": main()
